x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/FrozenIndexInputTests.java

```diff
@@ -44,7 +45,8 @@
     private static final ShardId SHARD_ID = new ShardId(new Index("_index_name", "_index_id"), 0);
 
     public void testRandomReads() throws IOException {
-        final String fileName = randomAlphaOfLength(5) + randomFileExtension();
+        // Can't test cfs files here since the randomization uses slice names with differing offsets which breaks the frozen cache
+        final String fileName = randomAlphaOfLength(5) + randomValueOtherThan(".cfs", ESIndexInputTestCase::randomFileExtension);
```

Comment:
No, the index input test uses the same cfs file with multiple nested file names reads using the same name but different slice lengths across calls. Since we now make some adjustments for the region layout of a file depending on its size, this causes breakage because a file that goes to the same cache key suddenly changes size. In the real world the cfs is final and this kind of size change wouldn't happen.